File: modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/model/Order.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.commerce.order.web.internal.model;

import com.liferay.commerce.frontend.model.LabelField;

/**
 * @author Alessio Antonio Rendina
 */
public class Order {

	public Order(
		String account, String accountCode, String amount, String channel,
		String createDateString, LabelField fulfillmentWorkflow, long orderId,
		LabelField orderStatus) {

		_account = account;
		_accountCode = accountCode;
		_amount = amount;
		_channel = channel;
		_createDateString = createDateString;
		_fulfillmentWorkflow = fulfillmentWorkflow;
		_orderId = orderId;
		_orderStatus = orderStatus;
	}

	public String getAccount() {
		return _account;
	}

	public String getAccountCode() {
		return _accountCode;
	}

	public String getAmount() {
		return _amount;
	}

	public String getChannel() {
		return _channel;
	}

	public String getCreateDateString() {
		return _createDateString;
	}

	public LabelField getFulfillmentWorkflow() {
		return _fulfillmentWorkflow;
	}

	public long getOrderId() {
		return _orderId;
	}

	public LabelField getOrderStatus() {
		return _orderStatus;
	}

	private final String _account;
	private final String _accountCode;
	private final String _amount;
	private final String _channel;
	private final String _createDateString;
	private final LabelField _fulfillmentWorkflow;
	private final long _orderId;
	private final LabelField _orderStatus;

}

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/enums/Order.ts
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export enum ORDER_WORKFLOW_STATUS_CODE {
	CANCELLED = 8,
	COMPLETED = 0,
	ON_HOLD = 20,
	PENDING = 1,
	PROCESSING = 10,
}

export enum ORDER_TYPES {
	DXPAPP = 'DXPAPP',
	CLOUDAPP = 'CLOUDAPP',
	SOLUTIONS7 = 'SOLUTIONS7',
	SOLUTIONS30 = 'SOLUTIONS30',
}
